jiant/utils/serialize.py

```diff
@@ -57,7 +58,48 @@ def bytes_to_float(b):
     return float(crc32(b) & 0xFFFFFFFF) / 2 ** 32
 
 
-def read_records(filename, repeatable=False, fraction=None):
+def read_records(filename, repeatable=False, fraction=1.0):
+    """Streaming read records from file.
+
+    Args:
+      filename: path to file of b64-encoded pickles, one per line
+      repeatable: if true, returns a RepeatableIterator that can read the file
+        multiple times.
+      fraction: if set to a float between 0 and 1, load only the specified percentage
+        of examples. Hashing is used to ensure that the same examples are loaded each
+        epoch.
+
+    Returns:
+      iterable, possible repeatable, yielding deserialized Python objects
+    """
+
+    if fraction < 1.0:
+        frac_filename = f"{filename}__fraction_{fraction}"
+        if not os.path.exists(frac_filename):
+            with open(filename, "rb") as fd:
+                examples = []
+                for line in fd:
+                    blob = base64.b64decode(line)
+                    hash_float = bytes_to_float(blob)
+                    if hash_float > fraction:
+                        continue
+                    example = pkl.loads(blob)
+                    examples.append(example)
+            write_records(examples, frac_filename)
+        filename = frac_filename
+
+    def _iter_fn():
+        with open(filename, "rb") as fd:
+            for line in fd:
+                blob = base64.b64decode(line)
+                example = pkl.loads(blob)
+                yield example
+
+    return RepeatableIterator(_iter_fn) if repeatable else _iter_fn()
+
+
+# temporary backup code, remove before merging
```

Comment:
Please remove.

Reply:
In case other reviewer would like to check this, I will remove it once the PR is approved